polkadot/xcm/pallet-xcm/src/lib.rs

```diff
@@ -1426,6 +1466,83 @@ pub mod pallet {
 				weight_limit,
 			)
 		}
+
+		/// Authorize another `aliaser` location to alias into the local `origin` making this call.
+		/// The `aliaser` is only authorized until the provided `expiry` block number.
+		///
+		/// Usually useful to allow your local account to be aliased into from a remote location
+		/// also under your control (like your account on another chain).
+		///
+		/// WARNING: make sure the caller `origin` (you) trusts the `aliaser` location to act in
+		/// their/your name. Once authorized using this call, the `aliaser` can freely impersonate
+		/// `origin` in XCM programs executed on the local chain.
+		#[pallet::call_index(14)]
+		pub fn add_authorized_alias(
+			origin: OriginFor<T>,
+			aliaser: Box<VersionedLocation>,
+			expires: Option<u64>,
+		) -> DispatchResult {
+			let origin: Location = T::ExecuteXcmOrigin::ensure_origin(origin)?;
+			let aliaser: Location = (*aliaser).try_into().map_err(|()| Error::<T>::BadVersion)?;
+			tracing::debug!(target: "xcm::pallet_xcm::add_authorized_alias", ?origin, ?aliaser, ?expires);
+			ensure!(origin != aliaser, Error::<T>::BadLocation);
+			if let Some(expiry) = expires {
+				ensure!(
+					expiry > frame_system::Pallet::<T>::block_number().saturated_into::<u64>(),
+					Error::<T>::ExpiresInPast
+				);
+			}
+			let versioned_origin = VersionedLocation::from(origin);
+			let versioned_aliaser = VersionedLocation::from(aliaser);
+			let mut authorized_aliases = AuthorizedAliases::<T>::get(&versioned_origin);
+			if authorized_aliases
+				.iter_mut()
+				.find_map(|aliaser| {
+					// if the aliaser already exists, just update its expiry block
+					if aliaser.location == versioned_aliaser {
+						aliaser.expiry = expires;
+						Some(())
+					} else {
+						None
+					}
+				})
+				.is_none()
+			{
+				// if the aliaser does not yet exist, add it to the list
+				let aliaser = OriginAliaser { location: versioned_aliaser, expiry: expires };
+				authorized_aliases
+					.try_push(aliaser)
+					.map_err(|_| Error::<T>::TooManyAuthorizedAliases)?;
+				// todo: hold storage deposit
```

Comment:
do you plan to address this todo in the current PR?

Reply:
just pushed something, want feedback before I clean it up all the way